app/one_click_unsubscribe/rest.py

```diff
@@ -52,3 +60,62 @@ def get_unsubscribe_request_data(notification, email_address):
         "service_id": notification.service_id,
         "email_address": email_address,
     }
+
+
+def create_unsubscribe_request_reports_summary(service_id):
+    reports_summary = []
+    batched_unsubscribe_reports_summary = []
+    unbatched_unsubscribe_report_summary = []
+    # Check for existing unsubscribe reports and create their summaries
+    if unsubscribe_request_reports := get_unsubscribe_request_reports_dao(service_id):
+        batched_unsubscribe_reports_summary = _create_batched_unsubscribe_request_reports_summary(
+            unsubscribe_request_reports,
+        )
+        batched_unsubscribe_reports_summary = batched_unsubscribe_reports_summary
+    # Check for unsubscribe requests and create a summary for them
+    if unbatched_unsubscribe_requests := get_unbatched_unsubscribe_requests_dao(service_id):
+        if batched_unsubscribe_reports_summary:
+            earliest_timestamp = batched_unsubscribe_reports_summary[0]["latest_timestamp"]
+        else:
+            service = dao_fetch_service_by_id(service_id)
+            earliest_timestamp = service.created_at
```

Comment:
I think it would be better if this was the timestamp of the oldest unbatched request. Otherwise it’s going to look confusing when there are dates from several years ago being displayed, which don’t then appear in the report

Reply:
Yeah, timestamp of the earliest unbatched request. I think the `created_at` date of the earliest request in the report makes more sense.